mindmeld/models/embedder_models.py

```diff
@@ -0,0 +1,185 @@
+        """
+        if os.path.exists(self.cache_path):
+            os.remove(self.cache_path)
+
+    def get_encodings(self, text_list):
+        """Fetches the encoded values from the cache, or generates them.
+        Args:
+            text_list (list): A list of text strings for which to get the embeddings.
+
+        Returns:
+            (list): A list of numpy arrays with the embeddings.
+        """
+        encoded = [self.cache.get(text, None) for text in text_list]
+        cache_miss_indices = [i for i, vec in enumerate(encoded) if vec is None]
+        text_to_encode = [text_list[i] for i in cache_miss_indices]
+        model_encoded_text = self.encode(text_to_encode)
+
+        for i, v in enumerate(cache_miss_indices):
+            encoded[v] = model_encoded_text[i]
+            self.cache[text_to_encode[i]] = model_encoded_text[i]
+        return encoded
+
+    def dump(self):
+        """Dumps the cache to disk.
+        """
+        with open(self.cache_path, "wb") as fp:
+            pickle.dump(self.cache, fp)
+
+
+class BertEmbedder(Embedder):
+    """
+    Encoder class for bert models as described here: https://github.com/UKPLab/sentence-transformers
+    """
+
+    def load(self, **kwargs):
+        DEFAULT_BERT = "bert-base-nli-mean-tokens"
+        if self.model_name == "default":
+            bert_model_name = DEFAULT_BERT
+            logger.info("No bert model specifications passed, using default.")
+        else:
+            bert_model_name = self.model_name
+        return SentenceTransformer(bert_model_name)
+
+    def encode(self, text_list):
+        return self.model.encode(text_list)
+
+
+class GloveEmbedder(Embedder):
+    """
+    Encoder class for GloVe embeddings as described here: https://nlp.stanford.edu/projects/glove/
+    """
+
+    def load(self, **kwargs):
+        DEFAULT_PADDING_LEN = 20
+        DEFAULT_EMBEDDING_DIM = 300
+        padding_length = kwargs.get("padding_length", DEFAULT_PADDING_LEN)
+        token_embedding_dimension = kwargs.get(
+            "token_embedding_dimension", DEFAULT_EMBEDDING_DIM
+        )
+        token_pretrained_embedding_filepath = kwargs.get(
+            "token_pretrained_embedding_filepath"
+        )
+        return WordSequenceEmbedding(
+            padding_length,
+            token_embedding_dimension,
+            token_pretrained_embedding_filepath,
+            use_padding=False,
+        )
+
+    def get_query_tokens(self, query):
+        """Splits into tokens, removes punctuation, and removes whitespace.
+        """
+        query = query.translate(str.maketrans("", "", string.punctuation))
+        tokens = query.split()
+        tokens = [t.strip() for t in tokens]
+        return tokens
+
+    def encode(self, text_list):
+        token_list = [self.get_query_tokens(text) for text in text_list]
+        vector_list = [self.model.encode_sequence_of_tokens(tl) for tl in token_list]
+        encoded_vecs = []
+        for vl in vector_list:
```

Comment:
Is this check necessary? Wont the axis make sure it leaves a list with a single list as is?

Reply:
It was converting a single list to a single numeric value rather than keeping the vector as is. Is there a way to force it to leave it as is?

Reply:
Shouldnt ` np.average(vl, axis=0) = vl` where len(vl) == 1?